scalardb/src/scalardb/db/cluster.clj

```diff
@@ -0,0 +1,220 @@
+(ns scalardb.db.cluster
+  (:require [clj-yaml.core :as yaml]
+            [clojure.string :as str]
+            [clojure.tools.logging :refer [info]]
+            [environ.core :refer [env]]
+            [jepsen
+             [control :as c]
+             [db :as db]]
+            [scalardb.nemesis.cluster :as n])
+  (:import (java.io File)))
+
+(def ^:private ^:const CLUSTER_VALUES_YAML "scalardb-cluster-custom-values.yaml")
+(def ^:private ^:const DEFAULT_CHAOS_MESH_VERSION "2.7.1")
+
+(def ^:private ^:const TIMEOUT_SEC 600)
+(def ^:private ^:const INTERVAL_SEC 10)
+
+(def ^:private ^:const CLUSTER_NODE_NAME "scalardb-cluster-node")
+
+(def ^:private ^:const CLUSTER_VALUES
+  {:envoy {:enabled true
+           :service {:type "LoadBalancer"}}
+
+   :scalardbCluster
+   {:image {:repository "ghcr.io/scalar-labs/scalardb-cluster-node"}
+
+    :scalardbClusterNodeProperties
+    (str/join "\n"
+              ;; ScalarDB Cluster configurations
+              ["scalar.db.cluster.membership.type=KUBERNETES"
+               "scalar.db.cluster.membership.kubernetes.endpoint.namespace_name=${env:SCALAR_DB_CLUSTER_MEMBERSHIP_KUBERNETES_ENDPOINT_NAMESPACE_NAME}"
+               "scalar.db.cluster.membership.kubernetes.endpoint.name=${env:SCALAR_DB_CLUSTER_MEMBERSHIP_KUBERNETES_ENDPOINT_NAME}"
+               ""
+               ;; Storage configurations
+               "scalar.db.storage=jdbc"
+               "scalar.db.contact_points=jdbc:postgresql://postgresql-scalardb-cluster.default.svc.cluster.local:5432/postgres"
+               "scalar.db.username=postgres"
+               "scalar.db.password=postgres"])
+
+    :imagePullSecrets [{:name "scalardb-ghcr-secret"}]}})
+
+(defn- install!
+  "Install prerequisites.
+  You should already have installed kind (or similar tool), kubectl and helm."
+  []
+  ;; postgre
+  (c/exec :helm :repo :add "bitnami" "https://charts.bitnami.com/bitnami")
+  ;; ScalarDB cluster
+  (c/exec :helm :repo :add
+          "scalar-labs" "https://scalar-labs.github.io/helm-charts")
+  ;; Chaos mesh
+  (c/exec :helm :repo :add "chaos-mesh" "https://charts.chaos-mesh.org")
```

Comment:
I’m not familiar with Chaos Mesh, but does it crash both the ScalarDB Cluster nodes and PostgreSQL?